### PR TITLE
Sm 101/feature/implement user change flow

### PR DESCRIPTION
**Pull request by andriibodryi**

### CODE REVIEW

File: src/components/CustomDrawerContent/CustomDrawerContent.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }, []);
+
+  useLogoutSubscription(navigateToAuthScreen);
+
+  const closeCustomerModal = useCallback(() => {
+    setCustomerModalState(false);
+  }, []);
+
+  const closeDrawerMenu = useCallback(() => {
+    props.navigation.closeDrawer();
```
Comment (Mykhailo-Kaduk) [inline src/components/CustomDrawerContent/CustomDrawerContent.tsx:38]: is it possible to use navigation from hooks instead of props?

Comment (Mykhailo-Kaduk) [reply src/components/CustomDrawerContent/CustomDrawerContent.tsx:38]: either way, please add navigation to callback deps

File: src/store/auth/sagas/logoutSaga.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        text: general.APPLY,
+        onPress: () => {
+          resolve(true);
+        },
+      },
+    ]);
+  });
+}
+
+export function* logoutSaga(): Generator {
```
Comment (Mykhailo-Kaduk) [inline src/store/auth/sagas/logoutSaga.ts:37]: please use SagaIterator from store instead of a Generator.

File: src/api/auth/userLogout.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { ApiResultType } from '~/api/requestTypes';
+import { LoginResponse } from './types';
+
+type UserDetailsRequest = () => Promise<ApiResultType<undefined>>;
+
+export const userLogout: UserDetailsRequest = () =>
+  new Promise((resolve) => {
+    const result: ApiResultType<LoginResponse> = {
```
Comment (artur-h) [inline src/api/auth/userLogout.ts:8]: I guess you meant here ApiResultType<LogoutResponse> instead of <LoginResponse>

**Review by Mykhailo-Kaduk — APPROVED** [review]

**Review by artur-h — APPROVED** [review]

**Pull request merged**